compiler/rustc_target/src/spec/mod.rs

```diff
@@ -479,6 +480,83 @@ macro_rules! supported_targets {
     };
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum StackProbeType {
+    /// Don't emit any stack probes.
+    None,
+    /// It is harmless to use this option even on targets that do not have backend support for
+    /// stack probes as the failure mode is the same as if no stack-probe option was specified in
+    /// the first place.
+    Inline,
+    /// Call `__rust_probestack` whenever stack needs to be probed.
+    Call,
+    /// Use inline option for LLVM versions later than specified in `min_llvm_version_for_inline`
+    /// and call `__rust_probestack` otherwise.
+    InlineOrCall { min_llvm_version_for_inline: (u32, u32, u32) },
+}
+
+impl StackProbeType {
+    fn from_json(json: &Json) -> Result<Self, String> {
```

Comment:
It's not clear to me -- is the target.json format supposed to be stable/compatible? i.e. Do we need to support someone who still has a boolean `stack_probes` setting?

Reply:
See https://github.com/rust-lang/rust/issues/71009